packages/InlineEditors/src/Editor/Editor.js

```diff
@@ -17,9 +51,34 @@ const Editor = React.forwardRef((props, ref) => {
   return isEditing ? (
     edit.props.children
   ) : (
-    <RawButton style={{ width: "100%" }} data-pka-anchor="inline-editing.raw-button" ref={ref} onClick={handleClick}>
-      {editorValue.props.children}
-    </RawButton>
+    <sc.Value status={status} data-pka-anchor="inline-editing.raw-button" ref={ref} onClick={handleClick}>
+      {status === statusTypes.ERROR ? (
+        <sc.Error>
+          <Tooltip Icon={ExclamationCircle} message={messageError} />
+        </sc.Error>
+      ) : null}
+      {status === statusTypes.LOADING && optimisticValue ? (
+        <sc.OptimisticValue>{optimisticValue}</sc.OptimisticValue>
+      ) : (
+        editorValue.props.children
+      )}
+
+      {status === statusTypes.IDLE ? (
+        <sc.Idle data-pka-anchor="inline-editors-table-edit-icon">
+          <Pencil />
+        </sc.Idle>
+      ) : null}
+      {status === statusTypes.LOADING ? (
+        <sc.Loading>
+          <Spinner size={Spinner.types.size.SMALL} />
+        </sc.Loading>
+      ) : null}
+      {status === statusTypes.SUCCEED ? (
+        <sc.Success onAnimationEnd={onAnimationEndSuccess}>
+          <Check />
+        </sc.Success>
+      ) : null}
```

Comment:
I usually prefer `switch () {case}` for the cases like these, but I guess up to you